components/brave_ads/browser/component_updater/resource_component.cc

```diff
@@ -0,0 +1,220 @@
+#include "brave/components/l10n/common/locale_util.h"
+
+namespace brave_ads {
+
+namespace {
+
+const uint16_t kCurrentSchemaVersion = 1;
+const char kSchemaVersionPath[] = "schemaVersion";
+const char kResourcePath[] = "resources";
+const char kResourceIdPath[] = "id";
+const char kResourceFilenamePath[] = "filename";
+const char kResourceVersionPath[] = "version";
+
+const char kComponentName[] = "Brave Ads Resources (%s)";
+
+const base::FilePath::CharType kManifestFile[] =
+    FILE_PATH_LITERAL("models.json");
+
+std::string GetIndex(std::string id, int version) {
+  return id + std::to_string(version);
+}
+
+}  // namespace
+
+ResourceComponent::ResourceComponent(Delegate* delegate)
+    : brave_component_updater::BraveComponent(delegate) {
+  DCHECK(delegate);
+}
+
+ResourceComponent::~ResourceComponent() = default;
+
+void ResourceComponent::RegisterComponentsForLocale(const std::string& locale) {
+  const std::string country_code = brave_l10n::GetCountryCode(locale);
+  RegisterComponentForCountryCode(country_code);
+
+  const std::string language_code = brave_l10n::GetLanguageCode(locale);
+  RegisterComponentForLanguageCode(language_code);
+}
+
+void ResourceComponent::AddObserver(Observer* observer) {
+  DCHECK(observer);
+
+  observers_.AddObserver(observer);
+}
+
+void ResourceComponent::RemoveObserver(Observer* observer) {
+  DCHECK(observer);
+
+  observers_.RemoveObserver(observer);
+}
+
+void ResourceComponent::NotifyObservers(const std::string& id) {
+  for (auto& observer : observers_) {
+    observer.OnResourceComponentUpdate(id);
+  }
+}
+
+base::Optional<base::FilePath> ResourceComponent::GetPath(const std::string& id,
+                                                          const int version) {
+  const std::string index = GetIndex(id, version);
+  const auto iter = resources_.find(index);
+  if (iter == resources_.end()) {
+    return base::nullopt;
+  }
+
+  const ResourceInfo resource = iter->second;
+  return resource.path;
+}
+
+//////////////////////////////////////////////////////////////////////////////
+
+void ResourceComponent::RegisterComponentForCountryCode(
+    const std::string& country_code) {
+  DCHECK(!country_code.empty());
+
+  const base::Optional<ComponentInfo> component =
+      GetComponentInfo(country_code);
+  if (!component) {
+    VLOG(1) << "no ads resources support for " << country_code;
+    return;
+  }
+
+  const std::string component_name =
+      base::StringPrintf(kComponentName, country_code.c_str());
+
+  VLOG(1) << "Registering " << component_name << " with id " << component->id;
+
+  Register(component_name, component->id, component->public_key);
+}
+
+void ResourceComponent::RegisterComponentForLanguageCode(
+    const std::string& language_code) {
+  DCHECK(!language_code.empty());
+
+  const base::Optional<ComponentInfo> component =
+      GetComponentInfo(language_code);
+  if (!component) {
+    VLOG(1) << "no ads resources support for " << language_code;
```

Comment:
nit: `Ads resource not supported for ...`

Reply:
fixed